src/App.jsx

```diff
@@ -14,45 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const GOOD_BY_LENGTH = 'length';
+const GOOD_BY_NAME = 'alphabetically';
+
+function getPreperedGoods(goods, sortGoods, isReverseGoods) {
+  const preperedGoods = [...goods];
+
+  if (sortGoods) {
+    preperedGoods.sort((good1, good2) => {
+      switch (sortGoods) {
+        case GOOD_BY_LENGTH:
+          return good1.length - good2.length;
+        case GOOD_BY_NAME:
+          return good1.localeCompare(good2);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverseGoods) {
+    preperedGoods.reverse();
+  }
+
+  return preperedGoods;
+}
+
+export const App = () => {
+  const [sortGoods, setSortGoods] = useState('');
+  const [isReverseGoods, setIsReversedGoods] = useState(false);
```

Comment:
'd' is redundant in setReverseGoods
```suggestion
  const [isReverseGoods, setIsReverseGoods] = useState(false);
```